app/precompiled.py

```diff
@@ -234,9 +225,55 @@ def overlay_template_areas(src_pdf, page_number=None, overlay=True):
     return png_from_pdf(pdf, page_number)
 
 
-def get_invalid_pages(src_pdf):
-    pdf_to_validate = _overlay_printable_areas(src_pdf)
-    return list(_get_out_of_bounds_pages(PdfFileReader(pdf_to_validate)))
+def get_invalid_pages_with_message(src_pdf):
+    message = ""
+    invalid_pages = []
+    are_pages_A4_portrait_oriented, invalid_pages = _are_pages_A4_portrait_oriented(src_pdf)
+    if not are_pages_A4_portrait_oriented:
+        message = 'The page orientation is landscape instead of portrait on '
+    else:
+        pdf_to_validate = _overlay_printable_areas(src_pdf)
+        invalid_pages = list(_get_out_of_bounds_pages(PdfFileReader(pdf_to_validate)))
+        if len(invalid_pages) > 0:
+            message = 'Content in this PDF is outside the printable area on '
+    if len(invalid_pages) > 0:
+        message += unescaped_formatted_list(
+            invalid_pages,
+            before_each='',
+            after_each='',
+            prefix='page',
+            prefix_plural='pages'
+        )
+    return (invalid_pages, message)
+
+
+def _are_pages_A4_portrait_oriented(src_pdf):
+    pdf = PdfFileReader(src_pdf)
+    invalid_pages = []
+    for page_num in range(0, pdf.numPages):
+        page = pdf.getPage(page_num)
+
+        # page size is in points, there are 72 points in an inch and 25.4 milimeters in an inch,
+        # hence the transformations below to get size in milimeters
+        page_height = float(page.mediaBox.getHeight()) / 72 * 25.4
+        page_width = float(page.mediaBox.getWidth()) / 72 * 25.4
+
+        # are pages A4 portrait oriented. For now we run this silently to check if our users' PDFs mostly comply.
+        # If we decide to fail PDFs that don't meet those conditions, we should probably also allow for
+        # height in range(209, 212) and width in range(296, 299) if page.get('/Rotate') in [90, 270]
+        # (that is a PDF that was originally landscape but has been rotated to be portrait-oriented)
+        if int(page_height) not in range(296, 299) or int(page_width) not in range(209, 212):
+            current_app.logger.warning('Letter size is not A4 on page {}, page size: {}x{}mm'.format(
+                page_num + 1, page_height, page_width
+            ))
+
+        # check if page orientation is not landscape:
+        if (
+            page_height < page_width and page.get('/Rotate') not in [90, 270]
+        ) or (page_height > page_width and page.get('/Rotate') not in [0, 180, None]):
+            invalid_pages.append(page_num + 1)
+            current_app.logger.warning("Letter landscape-oriented on page {}".format(page_num + 1))
```

Comment:
is it worth logging height, width and /rotate here?

Reply:
more info, sounds good 👍